to-do-list/src/components/UI/modal/MyModal.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import './MyModal.css'
+
+interface MyModalProps {
+    children: React.ReactNode,
+    visible: boolean,
+    setVisible: (value: boolean) => void;
+}
+
+const MyModal = ({children, visible, setVisible} : MyModalProps) =>{
+
+    const rootClasses = ['popup']
+    
+    if (visible){
+        rootClasses.push('visible');
+    }
+
+    return (
+        <div id="mypopup" className={rootClasses.join(' ')} onClick={() => setVisible(false)}>
```

Comment:
It's better to hide it from the parent component like {isShowModal && <CreateTaskModal ... />} to not to render component if it not needed